iroh-relay/src/server/client_conn.rs

```diff
@@ -384,6 +388,10 @@ impl Stream for RateLimitedRelayedStream {
         mut self: Pin<&mut Self>,
         cx: &mut std::task::Context<'_>,
     ) -> Poll<Option<Self::Item>> {
+        let Some(limiter) = self.limiter.clone() else {
```

Comment:
shouldn't the clone happen only on the branch that needs it?

Reply:
Good catch!  fixed.